### PR TITLE
Abstractify legacy conversion + conversion for SimpleRNN, GRU, LSTM

### PR DESCRIPTION
**Pull request by abhaikollara**
Abstract the process of creating legacy API interfaces.

### CODE REVIEW

**Comment by fchollet** [comment]
Why not generalize it to arbitrary numbers of positional arguments, in the same way? Just pass a list of args conversions and a list of kwargs conversions.

**Comment by abhaikollara** [comment]
@fchollet Working on it now

**Comment by abhaikollara** [comment]
@fchollet I'm a bit lost here, is there an arbitrary max number of positional arguments allowed for each layer ?

```python
raise TypeError("The `" + layer_name + '` can have at'
         'most one positional arguement (the `' + args_convert[1] +
          '` argument)')
```

**Comment by fchollet** [comment]
Basically, we want the user to be able to instantiate layers using the old API and using positional arguments, e.g. `Layer(a, b, c)`. Unless there are solid reasons (de-ambiguity) to limit this. Typically a reason to limit the use of positional arguments is that the order of the arguments has changed between Keras 1 and 2.

**Comment by abhaikollara** [comment]
@fchollet I've made the generalized version. The following layers pass tests as of now: `Dense` `Dropout` `SimpleRNN` `GRU` `LSTM` `MaxPooling1D` `AveragePooling1D`

**Comment by fchollet** [comment]
I think your approach is very good, but you are overcomplicating it.

Let's reframe the situation:

- Code using the old interface should work. This includes calling layers with positional arguments with the old interface.
- Code using the new interface should also work. This includes calling layers with positional arguments with the new interface.
- Sometimes we can't have both, because the order of the arguments has changed. In that case, we should forbid the use of positional arguments (beyond those that share the same order).

So basically we allow `args` to be passed only up to the point that argument order is the same in both APIs. We can achieve this by having a whitelist of allowed positional arguments.

I've just pushed a commit that refines the abstract interface.
